### PR TITLE
Update to chrono 0.4.34

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #.

# Rationale for this change
 

https://github.com/chronotope/chrono/pull/1328 changed chrono to no longer convert durations to a number of days when printing in the ISO compatible format. 

# What changes are included in this PR?

# Are there any user-facing changes?

Technically maybe, although it wasn't a breaking chrono release so I'm not sure...

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -115,6 +115,8 @@ jobs:
         uses: ./.github/actions/setup-builder
       - name: Install cargo-msrv
         run: cargo install cargo-msrv
+      - name: Downgrade arrow dependencies
+        run: cargo update -p ahash --precise 0.8.7
```
Comment (tustvold) [inline .github/workflows/rust.yml:119]: ahash 0.8.8 bumped the MSRV to 1.72

File: arrow-cast/src/display.rs [hunk 2]
```diff
@@ -1013,9 +1013,9 @@ mod tests {
         assert_eq!(pretty[2], "0 days 0 hours 0 mins 0.000001000 secs");
         assert_eq!(iso[3], "-PT0.000001S");
         assert_eq!(pretty[3], "0 days 0 hours 0 mins -0.000001000 secs");
-        assert_eq!(iso[4], "P45DT50554.123456789S");
+        assert_eq!(iso[4], "PT3938554.123456789S");
```
Comment (alamb) [inline arrow-cast/src/display.rs:1016]: why did these values change? They look closer to the pretty version with the former formatting (like 45 days)

Is this something we need to change in pretty print?

File: arrow-json/src/writer.rs [hunk 3]
```diff
@@ -1192,7 +1192,7 @@ mod tests {
 
         assert_json_eq(
             &buf,
-            r#"{"duration_sec":"PT120S","duration_msec":"PT0.120S","duration_usec":"PT0.000120S","duration_nsec":"PT0.000000120S","name":"a"}
+            r#"{"duration_sec":"PT120S","duration_msec":"PT0.12S","duration_usec":"PT0.00012S","duration_nsec":"PT0.00000012S","name":"a"}
```
Comment (alamb) [inline arrow-json/src/writer.rs:1195]: this looks better

**Review by alamb — COMMENTED** [review]
Thanks @tustvold

File: arrow-cast/src/display.rs (hunk 2 above)
Comment (tustvold) [reply arrow-cast/src/display.rs:1016]: As per the PR description - https://github.com/chronotope/chrono/pull/1328 changed the formatting to be more "correct"

Comment (alamb) [reply arrow-cast/src/display.rs:1016]: Makes sense

**Review by alamb — APPROVED** [review]

**Pull request merged**